pkg/client/restclient/request.go

```diff
@@ -749,6 +749,16 @@ func (r *Request) Stream() (io.ReadCloser, error) {
 
 		result := r.transformResponse(resp, req)
 		if result.err != nil {
```

Comment:
Actually don't want the if block - Error() can be dynamically constructed, so this could be:

```
result := r.transformResponse(res, req)
err := result.Error()
if err == nil {
  err = fmt.Errorf("%d while accessing %v: %s", result.statusCode, url, string(result.body))
}
return err
```

That ensures that if err was nil but body was set, we can decode an error out.